### PR TITLE
Templating changes from the opa-operator user-info-fetcher work

### PR DESCRIPTION
**Pull request by NickLarsenNZ**
Templating changes from https://github.com/stackabletech/opa-operator/pull/433#pullrequestreview-1696890759

### CODE REVIEW

File: template/deploy/helm/[[operator]]/templates/deployment.yaml.j2 [hunk 1]
```diff
@@ -17,8 +17,11 @@ spec:
     metadata:
       {{- with .Values.podAnnotations }}
       annotations:
+        internal.stackable.tech/image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
```
Comment (NickLarsenNZ) [inline template/deploy/helm/[[operator]]/templates/deployment.yaml.j2:20]: Looks like the value files are defined in the operator repos, and I noticed `image.tag` is not defined in opa-operator. It works there, so I don't see an issue here.

File: template/deploy/helm/[[operator]]/templates/deployment.yaml.j2 [hunk 2]
```diff
@@ -17,8 +17,11 @@ spec:
...
+        {{- with .Values.podAnnotations }}
         checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
         {{- toYaml . | nindent 8 }}
+        {{- end }}
```
Comment (NickLarsenNZ) [inline template/deploy/helm/[[operator]]/templates/deployment.yaml.j2:24]: @nightkr, I'm unsure if this change was supposed to come across too.

**Review by sbernauer — CHANGES_REQUESTED** [review]

File: template/deploy/helm/[[operator]]/templates/deployment.yaml.j2 [hunk 3]
```diff
@@ -17,8 +17,11 @@ spec:
     metadata:
       {{- with .Values.podAnnotations }}
       annotations:
+        internal.stackable.tech/image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+        {{- with .Values.podAnnotations }}
```
[the remark this answers is not in the document]
Comment (NickLarsenNZ) [reply template/deploy/helm/[[operator]]/templates/deployment.yaml.j2:20]: Done in 36c2c3d

File: template/deploy/helm/[[operator]]/templates/deployment.yaml.j2 (hunk 2 above)
Comment (NickLarsenNZ) [reply template/deploy/helm/[[operator]]/templates/deployment.yaml.j2]: Resolved in 36c2c3d

File: template/deploy/helm/[[operator]]/templates/deployment.yaml.j2 (hunk 1 above)
Comment (sbernauer) [reply template/deploy/helm/[[operator]]/templates/deployment.yaml.j2:19]: Checking other operators, they don't specify image.tag either, so looks good :+1:

Comment (nightkr) [reply template/deploy/helm/[[operator]]/templates/deployment.yaml.j2:19]: Aye, that's what the `default` clause is for ^^

**Pull request merged**